paasta_tools/kubernetes_tools.py

```diff
@@ -340,6 +341,9 @@ def copy(self) -> "KubernetesDeploymentConfig":
             soa_dir=self.soa_dir,
         )
 
+    def get_kubernetes_namespace(self) -> str:
+        return KUBERNETES_NAMESPACE
```

Comment:
I hope not? For now I think this is the right place, I thought about setting it on the client but I think it would need bigger changes because the same client might have to deal with multiple namespaces...

Reply:
what if we could make `kube_client.with(namespace: ns)` return a copy of the client object instead of mutating the original?